### PR TITLE
Feature/different variables

### PR DESCRIPTION
**Pull request by Rambatino**

### CODE REVIEW

File: CHAID/CHAID.py [hunk 1]
```diff
@@ -95,6 +97,11 @@ def __missing__(self, key):
         return value
 
 
+class CHAIDVariableTypes(Enum):
```
Comment (xulaus) [inline CHAID/CHAID.py]: Needs a doctoring

File: CHAID/CHAID.py [hunk 2]
```diff
@@ -119,11 +126,14 @@ class CHAID(object):
     split_titles : array-like 
         array of names for the independent variables in the data
     """
```
Comment (xulaus) [inline CHAID/CHAID.py]: Add new args to doc string

File: CHAID/CHAID.py [hunk 3]
```diff
@@ -137,7 +147,8 @@ def __init__(self, ndarr, arr, alpha_merge=0.05, max_depth=2, min_sample=30, mis
         self.node(np.arange(0, self.data_size, dtype=np.int), ndarr, arr)
 
     @staticmethod
-    def from_pandas_df(df, i_variables, d_variable, alpha_merge=0.05, max_depth=2, min_sample=30):
+    def from_pandas_df(df, i_variables, d_variable, alpha_merge=0.05, max_depth=2, min_sample=30,
+                        independent_variable_types=None, dependent_variable_type=None):
         """
         Helper method to pre-process a pandas data frame in order to run CHAID analysis
```
Comment (xulaus) [inline CHAID/CHAID.py]: Add new args to doc string

File: CHAID/CHAID.py [hunk 4]
```diff
@@ -159,7 +170,13 @@ def from_pandas_df(df, i_variables, d_variable, alpha_merge=0.05, max_depth=2, m
         ind_df = df[i_variables]
         ind_values = ind_df.values
         dep_values = df[d_variable].values
-        return CHAID(ind_values, dep_values, alpha_merge, max_depth, min_sample, split_titles=list(ind_df.columns.values))
+        if independent_variable_types is not None:
+            independent_variable_types = [CHAIDVariableTypes[x] for x in independent_variable_types]
```
Comment (xulaus) [inline CHAID/CHAID.py]: Should this (`CHAIDVariableTypes[x]`) really be square brackets? Also piping in numbers seems a bit icky. Why not just make the developer either use the enum, or cast from string. Neither are neat but they are explicit.

File: CHAID/CHAID.py [hunk 5]
```diff
@@ -159,7 +170,13 @@ def from_pandas_df(df, i_variables, d_variable, alpha_merge=0.05, max_depth=2, m
...
+        if dependent_variable_type is not None:
+            dependent_variable_type = CHAIDVariableTypes[dependent_variable_type]
```
Comment (xulaus) [inline CHAID/CHAID.py]: Same comment about square brackets

File: CHAID/CHAID.py [hunk 6]
```diff
@@ -248,35 +268,57 @@ def generate_best_split(self, ind, dep):
...
 
             while len(unique) > 1:
                 size = int((len(unique) * (len(unique) - 1)) / 2)
                 sub_data = np.ndarray(shape=(size, 3), dtype=object, order='F')
-                for j, comb in enumerate(it.combinations(unique, 2)):
+
+                if self.i_v_types[i] == CHAIDVariableTypes(0):
+                    combinations = it.combinations(unique, 2)
+                else:
+                    combinations = list(it.combinations(unique, 2))
```
Comment (xulaus) [inline CHAID/CHAID.py]: `combinations = ( x for x in zip(unique, unique[:-1]) if x[0] in mappings)`?

Comment (xulaus) [inline CHAID/CHAID.py]: ` for k in unique: combinations += [(min(mappings[k]) - 1, max(mappings[k])), (min(mappings[k]), max(mappings[k])+1)]`

File: CHAID/__main__.py [hunk 7]
```diff
@@ -5,18 +5,19 @@
...
 def main():
     """Entry point when module is run from command line"""
-
     parser = argparse.ArgumentParser(description='Run the chaid algorithm on a csv file.')
-    parser.add_argument('file')
-    parser.add_argument('dependent_variable', nargs=1)
-    parser.add_argument('independent_variables', nargs='+')
-
+    parser.add_argument('--csv')
+    parser.add_argument('--dependent-variable', help='The name of the dependent variable')
```
Comment (xulaus) [inline CHAID/__main__.py]: where has the `nargs` argument gone? I think it is still needed

**Comment by xulaus** [comment]
I think the logic for the variable type stuff should be able to be extracted into separate classes some how. Maybe a chaid column that handles grouping and unique values?

**Comment by Rambatino** [comment]
I'm going to close this as it has diverged substantially from the main branch

**Pull request closed without merging**